### ISSUE TITLE
Segmentation fault with PHP 8.2.0RC4 and current xhprof master branch content 

### ISSUE THREAD

**Issue by llaville** [longxinH/xhprof#71]
## Bug Report

1. Script to test

```php
<?php
class SqlitePlatform
{
    public static function udfSqrt($value)
    {
        return sqrt($value);
    }
}

$userDefinedFunctions = [
    'sqrt' => ['callback' => [SqlitePlatform::class, 'udfSqrt'], 'numArgs' => 1],
];

$dsn = "sqlite:user-db.sqlite";
try {
    $pdo = new PDO($dsn);
    foreach ($userDefinedFunctions as $fn => $data) {
        $pdo->sqliteCreateFunction($fn, $data['callback'], $data['numArgs']);
    }
    \var_dump($pdo);

} catch (PDOException $e) {
    \var_dump($e);
}
```

2. What did you expect to see

```
object(PDO)#1 (0) {
}
```

3. What did you see instead

```
Segmentation fault
```

4. What is your Xhprof version

`2.3.7` and current master branch too (commit a7befadae5b57bb63dff0cfaf05aa642e564aa03)

I've though first that it was a PHP regression (see report https://github.com/php/php-src/issues/9749) until I found that it was this extension that run me in trouble.

BTW, run version 2.3.7 run fine with PHP 8.2.0RC3 (no segmentation fault)

**Comment by cmb69** [comment]
FWIW, this issue has been introduced by https://github.com/php/php-src/commit/9be00e3935b69930b5c2603e78f5e51026102377.

**Comment by longxinH** [comment]
@llaville Has the problem been resolved?

**Comment by llaville** [comment]
@longxinH No. This is the reason why I've opened this issue.

**Comment by llaville** [comment]
FYI: Issue still exits with PHP 8.2.0RC5 and xhprof 2.3.8

**Comment by longxinH** [comment]
Does this problem only occur in RC4? I did not reproduce in RC1

**Comment by longxinH** [comment]
This bug should be fixed in RC6 [rc6](https://github.com/php/php-src/commit/56c121cea2fa51533ebb5fc4cb4ac71324ffd1e3)

**Comment by llaville** [comment]
Will check it ASAP, when RC6 will be released !

**Comment by llaville** [comment]
@longxinH I confirmed it's fixed with 8.2.0RC6 and latest release 2.3.8
